src/org/labkey/test/components/ui/entities/ParentEntityEditPanel.java

```diff
@@ -8,80 +8,114 @@
      * @param element The WebElement containing the panel.
      * @param driver A reference to the WebDriver
-     * @param parentType An enum of type @ParentType used to identify the type of lineage entity is being edited.
-     *                   This is used to set the expected text when searching for web elements.
      */
-    public ParentEntityEditPanel(WebElement element, WebDriver driver, ParentType parentType)
+    public ParentEntityEditPanel(WebElement element, WebDriver driver)
     {
-        _driver = driver;
-        _editingDiv = element;
-        _parentType = parentType;
+        this.driver = driver;
+        editingDiv = element;
     }
 
     @Override
     public WebElement getComponentElement()
     {
-        return _editingDiv;
+        return editingDiv;
     }
 
     @Override
     protected WebDriver getDriver()
     {
-        return _driver;
+        return driver;
     }
 
-    // TODO: It would be nice if the save and cancel buttons would return a specific page type. That would require
-    //  using generics (much in the same way the modal dialog does), but I don't have time to do that right now.
-
-    /**
-     * Check to see if the panel is loaded. A panel is considered loaded if all of the combo-boxes in the panel are
-     * valid and are not loading.
-     *
-     * @return True if all of the combo-box references are valid and not loading.
-     */
-    public boolean isPanelLoaded()
+    @Override
+    protected void waitForReady()
     {
-        return getAllTypeCombo().stream().allMatch(rs -> rs.isInteractive() && !rs.isLoading());
+
+        // The panel is ready if:
+        // 1. There are no spinners.
+        // 2. If there are parents (i.e. FilterReactSelect controls present) they should all be interactive (not in the process of loading).
+        // 3. If there are no parents there should be only one entity type select (ReactSelect) and it should be interactive and have a list of options.
+        WebDriverWrapper.waitFor(()->
+        {
+            if(BootstrapLocators.loadingSpinner.findWhenNeeded(this).isDisplayed())
+            {
+                return false;
+            }
+            else
+            {
+
+                try
+                {
+
+                    // If there are existing parents then all the FilteringReactSelect controls (i.e. the parents) should
+                    // be interactive. When this panel is in a dialog waitForReady can be called after the parent entity type is selected.
+                    // In that scenario the parent selector will be empty, but it should be interactive.
+                    List<FilteringReactSelect> filteringReactSelects = FilteringReactSelect.finder(getDriver()).findAll(this);
+
+                    if(filteringReactSelects.size() > 1)
+                    {
+                        for (FilteringReactSelect select : filteringReactSelects)
+                        {
+                            if (!select.isInteractive())
```

Comment:
Should wait for selections as well.
```suggestion
                            if (!select.isInteractive() || !select.hasSelection())
```